calibrators/docs/tutorials/tag_based_pnp_calibrator.md

```diff
@@ -0,0 +1,157 @@
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/lidar_detection.jpg"  alt="lidar_detection">
+</p>
+
+### Apriltag (Lidartag)
+
+Details can be found in the [base documentation](../../tag_based_pnp_calibrator/README.md)
+
+## Launching the tool
+
+In this tutorial, we take the X2 project as an example.
+First, run the sensor calibration manager:
+
+```text
+ros2 run sensor_calibration_manager sensor_calibration_manager
+```
+
+In `project`, select `x2`, and in `calibrator`, select `tag_based_pnp_calibrator`. Then, press `Continue`.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/menu1.jpg"  alt="menu1">
+</p>
+
+A menu titled `Launcher configuration` should appear in the UI, and the user may change any parameter he deems convenient.
+For this tutorial, we modify the default values `calibration_pairs` from `9` to `8` as the rosbag has 8 apriltag detections and also modify the `camera_name` from `camera0` to `camera6`. After configuring the parameters, click `Launch`.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/menu2.jpg"  alt="menu2">
+</p>
+
+The following UI should be displayed. When the `Calibrate` button becomes available, click it.
+If it does not become available, it means that either the required `tf` or services are not available. In this case, since the `tf` are published by the provided rosbag, run the bag with the command `ros2 bag play camera_lidar.db3 --clock -r 0.5` and click the `Calibrate` button.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/menu3.jpg"  alt="menu3">
+</p>
+
+## Calibration
+
+The calibration starts automatically after clicking the `Calibrate` button. It will keep calibrating the lidartag detections and apriltag detections until the number of the detections fits the user-defined `calibration_pairs` in the `Launcher configuration`.
+
+When the user starts the calibration, `rviz` and the `image view` should be displayed like below.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization1.jpg"  alt="visualization1">
+</p>
+
+After the tools detect the lidartag and apriltag, it will show the detection markers on the `rviz` and the `image view`. The text in rviz will also display the number of pairs of lidartag/apriltag detections.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization2.jpg"  alt="visualization2">
+</p>
+
+Once the user observes the number of pairs increases in the `rviz`, he can start moving the tag to another position.
+
+While moving the tag to another position, the user should move the tag decisively from one place to the other, not stopping and starting mid-way.
+Please make sure the new position is at least `calibration_min_pair_distance` from the closest detection in the set of converged detections and the tag is in the FOV of both lidar and camera.
```

Comment:
```suggestion
Please make sure that the new position is at least `calibration_min_pair_distance` from the closest detection in the set of converged detections and that the tag is in the FOV of both lidar and camera.
```

Reply:
The commit suggestion is not in the bottom part of the conversation, so I modified it directly. Otherwise, I would need to expand all of the conversation again. 
fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/46fc240fd398ef0207b480690067faccfb31e6b5